### ISSUE TITLE
Cookie & session data needs to carry across subdomains.

### ISSUE THREAD

**Issue by LaustinSpayce** [Talishar/Talishar#145]
So I tried chatting on the new fe and on the be, turns out the cookies / sessions are not transferred across subdomains, there seems to be some php.ini file and some php stuff to fix this. So it'll need fixing. Else we end up with like this situation:
<img width="517" alt="image" src="https://user-images.githubusercontent.com/5060535/211188878-aad43b0b-07b1-4d9f-9805-c88dd6a8c9dc.png">

**Comment by LaustinSpayce** [comment]
So a very quick note we can try changing some php.ini settings to have session data persist across subdomains:
https://stackoverflow.com/questions/1064243/php-sessions-across-sub-domains

and areas needing updating could include:
https://github.com/Talishar/Talishar/blob/08b3ed518613190f6c26f494739b43edc96645b6/JoinGameInput.php#L482-L492

And affecting other places where `session_start()` is invoked. But also php.ini so it'll need adjusting in prod and beta

**Comment by LaustinSpayce** [comment]
Some reference:

Chat curl from FE:
```
curl 'https://talishar.net/game/SubmitChat.php?gameName=515826&playerID=1&chatText=Test%20Chat&authKey=8ac921a69f3ee915f710a3da7a1b41330bce0d5df81044341c57b82a8f103b76' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:108.0) Gecko/20100101 Firefox/108.0' -H 'Accept: */*' -H 'Accept-Language: en-GB,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br' -H 'Connection: keep-alive' -H 'Referer: https://talishar.net/game/NextTurn4.php?gameName=515826&playerID=1' -H 'Cookie: __gads=ID=4824d6747fab9da8-22ff587b21d800e8:T=1667811346:RT=1667811346:S=ALNI_MZxh3pKDwHtboaXNJMeoxhwJXk3CA; __gpi=UID=00000b771b9a8724:T=1667811346:RT=1669619514:S=ALNI_MbL4SGyyB8elG2zGOgh67bar91KCA; rememberMeToken=1c4f1bd057fc3f11e0d4426c862a7d3b828901c8163edb2bd3bab044d384b274; PHPSESSID=pbrj2saqho9qdf4cnsgetogu1k; lastAuthKey=8ac921a69f3ee915f710a3da7a1b41330bce0d5df81044341c57b82a8f103b76' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-origin' -H 'TE: trailers'
```

Test curl from React:
```
curl 'https://talishar.net/game/SubmitChat.php?gameName=515826&playerID=1&authKey=8ac921a69f3ee915f710a3da7a1b41330bce0d5df81044341c57b82a8f103b76&chatText=Test+Chat' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:108.0) Gecko/20100101 Firefox/108.0' -H 'Accept: */*' -H 'Accept-Language: en-GB,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br' -H 'Referer: https://fe.talishar.net/' -H 'Origin: https://fe.talishar.net' -H 'Connection: keep-alive' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-site' -H 'TE: trailers'
```

We need to double check `Submitchat.php` too because react FE is getting CORS Cross Origin headers error.

Anyway as you can see all the php session cookies are for talishar.net and not .talishar.net

<img width="796" alt="Screenshot 2023-01-10 at 2 09 40 PM" src="https://user-images.githubusercontent.com/5060535/211474855-3cf5ec52-7870-4f1f-84bc-c1e9cce9f205.png">

**Comment by OotTheMonk** [comment]
I tried doing the recommendation from that site, but it ended up breaking the game entirely. From further research, I found that it's recommended to set the session domain in the php.ini file instead. I tried doing that, and reloading the webserver, but it doesn't seem to have fixed the problem.
@LaustinSpayce how did you view all the session in your browser (previous issue)
@Bluffkin do you know how to solve this one? Did I set the domain in the php.ini file incorrect? Or maybe it needs a full shutdown and restart?
This is the line I added to:
session.cookie_domain=.talishar.net
Also tried without the leading dot:
session.cookie_domain=talishar.net

**Comment by LaustinSpayce** [comment]
To see your browsers current cookies:
On firefox, press F12 -> storage -> cookies -> site
On chrome it would be F12 -> application -> storage -> cookies -> site

**Comment by LaustinSpayce** [comment]
Try moving this line up to just after the includes
https://github.com/Talishar/Talishar/blob/3e4e2ead90bb34a3c4e2d263e025a9dfed269cef/SubmitChat.php#L24

**Comment by OotTheMonk** [comment]
Moving that line fixes the CORS error, but chat in the new FE seems to still not work. My best guess is that it's still not finding the session, but I'm not sure how to debug that.

**Comment by OotTheMonk** [comment]
![image](https://user-images.githubusercontent.com/77394768/211952669-90dc2c8b-dc30-4f08-b2ce-3dfdce09c5cb.png)
This is fixed 807f0e1e
